silq/instrument_interfaces/keysight/E8267D_interface.py

```diff
@@ -671,7 +695,7 @@ def get_additional_pulses(self, interface: InstrumentInterface):
                             'input_instrument': interface.instrument_name(),
                             'input_channel': 'trig_in'})]
 
-        assert (interface.IQ_modulation() == 'on') and \
+        assert (IQ_modulation == 'on') and \
                (interface.FM_mode() == 'IQ'), 'FM_mode should be IQ and IQ_modulation should be ON ' \
```

Comment:
Is FM_mode part of `frequency_settings`? Or can you only access this via the interface?
I'm a bit out of the loop, I'm not sure when the interface was changed to pass more variables than just the interface...

Reply:
No, so Serwan decided to detach some of the settings from the interface to make functions more independent, but not everything is in there, like FM_mode, etc. However, the interface still contains all of the parameters.

Reply:
Righto, I guess that's something that should be consolidated in future.